### ISSUE TITLE
warning: Unresolved var: gen/fmap in cljc files

### ISSUE THREAD

**Issue by stijnopheide** [clj-kondo/clj-kondo#2384]
- [X] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

`2024.08.01`

**platform**

JVM and JS

**problem**

In a .cljc file, clojure specs gen/fmap is indicated as an unresolved var.

**repro**

file `break.clj`
```clojure
(ns break
  (:require
    [clojure.spec.alpha :as s]
    #?(:clj  [clojure.spec.gen.alpha :as gen]
       :cljs [cljs.spec.gen.alpha :as gen])))

(s/def ::break
  (s/with-gen
    nat-int?
    #(gen/fmap
       inc
       (s/gen integer?))))
```

run clj-kondo
```bash
clj-kondo --lint break.cljc
break.cljc:10:7: warning: Unresolved var: gen/fmap
linting took 15ms, errors: 0, warnings: 1
```

**config**

default

**expected behavior**

No warning is issued.
